New collaborative commentary

angerman opened the pull request:
This proposal tries to solve most of the issues raised over the few days on the ghc-dev mailing list regarding the wiki.

[Rendered](https://github.com/ghc-megacorp/ghc-proposals/blob/collaborative-commentary/proposals/0000-new-collaborative-commentary.rst)

angerman committed:
Create 0000-new-collaborative-commentary.rst

initial checkin.

hvr commented:
on #ghc it was mentioned that the LLVM project has (http://llvm.org/docs/), which is also sphinx generated and is one direction we could pursue for GHC as well

Commit:
Spelling, typos, etc.

Commit:
Introduce two possible alternatives

The second alternative essentially makes the migration of wiki content optional. This, I believe, is more realistic.

goldfirere commented:
I've just submitted a PR against this PR, as my edits make largish changes to the document and I didn't want to step on any toes. You can view this rendered [here](https://github.com/ghc-megacorp/ghc-proposals/blob/partial-migration/proposals/0000-new-collaborative-commentary.rst).

In summary: I wish the wiki migration to be optional, mostly because I doubt we will find the labor to make it mandatory, and I don't want to promise something we can't do.

bgamari commented:
I would certainly be fine with moving in this direction. I have long advocated moving content out of the Wiki to proper forms of version control and I don't think anything about Trac makes it uniquely qualified to handle our commentary.

I do wonder, however, whether this change will have any positive on the maintenance of the Commentary that we have; I have a feeling the problem is more that we have a general lack of knowledge than a lack of willing authors.

gridaphobe commented on proposals/0000-new-collaborative-commentary.rst:
```diff
@@ -0,0 +1,55 @@
+Unresolved Questions
+--------------------
+
+None yet.
```
One thing that's **nice** about the trac wiki is that the transition from viewing to editing is seamless, you just have to click the "edit" button. It would be nice to preserve this feature, e.g. if we were hosting the docs on github, each page could contain a link to edit the source on github.

goldfirere commented:
GitHub has an pencil icon when viewing pages that allow you to edit and either commit (if you have access rights) or submit a PR. It's really quite convenient. The existence of this feature is one of the reasons I back this general direction.

Commit:
Remove reference to markdown.

gridaphobe commented:
@goldfirere Indeed, I'm just suggesting that whatever we do in terms of formatting/hosting the wiki, it should include a link to the GitHub editor for each page :)

cartazio commented:
At the end of the day.... All that matters is that it's easy to read and
engage with.

I will confess that I'm finding my self much less apt to read the proposal
discussions because the email thread doesn't include the initial proposal.

In contrast , track sends out each revision of a ticket.

This doesn't quite mean it's a critique.  But one dimension that's not been
discussed is the intrinsic signal to noise ratio of different types of
subscribing to content changes.

Will we seek to have owner listeners for different parts of the wiki of GHC
lore? Perhaps we should?

On Sep 29, 2016 2:39 PM, "Eric Seidel" <notifications@github.com> wrote:

> @goldfirere <https://github.com/goldfirere> Indeed, I'm just suggesting
> that whatever we do in terms of formatting/hosting the wiki, it should
> include a link to the GitHub editor for each page :)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ghc-proposals/ghc-proposals/pull/10#issuecomment-250554448>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAQwpZbIwR6hQwA0i0WOLQ5_4U-LkC8ks5qvAX0gaJpZM4KJsJ1>
> .
>

Commit:
Update 0000-new-collaborative-commentary.rst

Commit:
Merge pull request #3 from ghc-megacorp/partial-migration

Introduce two possible alternatives

takenobu-hs commented:
About "remaining fibers".

Richard said on [devs ML](https://mail.haskell.org/pipermail/ghc-devs/2016-September/012924.html),
> So I'm at a loss of what to do about these remaining fibers. Concrete suggestions, anyone?

As Richard pointed about remaining fibers:
(1) search engines still find out-of-date documentation
(2) the wiki is not discoverable


My concrete suggestion is here:

For (1):
When we find out-of-date documentation, we directly modify head of the document.
We manually insert a comment like "This content is out-of-date for GHC8.0".
(New contributors easy can do it.)

For (2):
We provide manually-written multiple indexes for different views on a portal page.
Contributors could maintain each index themselves.
(New comers will choose his favorite index for his exploring.)

Furthermore, we provide a simple search box for multiple wiki sites.
(Please wait for a while. I'll prepare simple-conceptual demonstration with web.)

[1]: https://mail.haskell.org/pipermail/ghc-devs/2016-September/012924.html

bgamari committed:
Fix indentation

bgamari commented on proposals/0000-new-collaborative-commentary.rst:
```diff
@@ -0,0 +1,73 @@
+GitHub and not on the wiki anyway. This backports the problem. In some years, however,
+I would expect most features to be on GitHub and this drawback to diminish.
+
+* Loss of Trac integration with comments; it's now harder to link to Trac tickets.
```
This needn't necessarily be the case given a sufficiently powerful markup language and processing pipeline. If we would use Sphinx we could easily replicate the `:ghc-ticket:` directive that we currently have in the users guide. It's not quite as short as just writing `#1234`, but it's certainly better than writing a URL.

bgamari commented:
Takenobu Tani <notifications@github.com> writes:

> About "remaining fibers".

snip

> My concrete suggestion is here:

I'm a bit concerned that this suggestion may prove ineffective in
practice; there is nothing stopping contributors from manually inserting
"this is out-of-date" tags and maintaining multiple indexes today, but
they don't.

Perhaps the problem really is that contributors haven't realized that
this would be helpful, but I have my doubts.

angerman committed:
Merge pull request #4 from ghc-megacorp/bgamari-patch-1

Fix indentation

takenobu-hs commented:
@bgamari Indeed.
Even now, it has not been stopped. Umm.

They might have a psychological barrier to update documents...
I'll think how to remove the barriers.

A simple idea,
we recommend the update of the documents at a front page of a wiki.
Especially, about updating outdated pages and construct indexes.

However, it is not yet systematic...

goldfirere commented:
The problem here is getting visitors to follow @takenobu-hs 's good suggestions. Is it possible in Trac to get a banner that displays whenever a page has not been updated for some time? Then a reader will at least know to tread carefully. (Even better would be the ability to, say, change the page background...)

bgamari commented:
Richard Eisenberg <notifications@github.com> writes:

> The problem here is getting visitors to follow @takenobu-hs 's good
> suggestions. Is it possible in Trac to get a banner that displays
> whenever a page has not been updated for some time? Then a reader will
> at least know to tread carefully. (Even better would be the ability
> to, say, change the page background...)
>
Well, anything is in principle possible. However, I believe this would
require hacking our Trac installation, so there is the question of cost
versus benefits.

takenobu-hs commented:
> Furthermore, we provide a simple search box for multiple wiki sites.
> (Please wait for a while. I'll prepare simple-conceptual demonstration with web.)
	
I pushed my simple-conceptual demonstration web to search in multiple wiki sites.
[Rapid prototyping is here.](https://takenobu-hs.github.io/haskell-wiki-search)

michalt commented:
+1 from me on moving the GHC commentary to GitHub and reStructuredText. I really prefer to send PRs than edit the wiki (due to reviews).

angerman commented:
The proposal contains three larger subsets as I see right now:
- migrating Trac to Phabricator. I think @mpickering has some interest in tackling this.
- Moving the Documentation/Commentary part from the Wiki into an rST base repository (+potentially Sphinx output somehwere)
- A better new contributors guide (with a sample onboarding document)

I'm not really sure how to solve the issue @cartazio raised wrt to the SNR. You'd probably have to subscribe to the repository to get notified about changes to the docs/commentary, which could again be pretty high SNR.

I 'like @takenobu-hs conceptual demonstration, and think it's a great add-on for the current situation, improving a lot over tracs built in search. However, it adds yet another "tool/website",
maybe that could become part of a better new contributors guide?

I'm planning to break this proposal into the mention sub proposals over the week and see if I forgot something. If someone else want's to take over one of those proposals (please!), let me know.

Cheers,
 Moritz

goldfirere commented:
I'm sorry, but I'm a bit confused by @angerman 's most recent comment. Are you combining this PR with #9 ? This PR seems only to deal with the second bullet point.

Sorry if a comment like this is simply nit-picking, but there's a *lot* going on in these PRs and I think it's important to stay organized. And, thanks, @angerman , for the time you're putting into moving all of this forward!

takenobu-hs commented:
@angerman Thanks. 
If it would be useful for the current situation, it may be easy to import my html source to any web pages.
Because, I used only simple tags(form, input, a) and small javascript code.
No need special tool and/or special website :)

bgamari commented:
Hi @angerman,

This proposal has been open for several months without any additional discussion. What would you like to do with this?

angerman commented:
Closing — this has been dormant since 2019.
